operations_on_datastructures/trie_multiple_search.cpp

```diff
@@ -0,0 +1,471 @@
+        int letter_index = tolower(delete_string[remove_index]) - 97;
+
+        DeleteFrom(delete_from->english[letter_index], delete_string,
+                   remove_index + 1);
+
+        delete delete_from;
+    }
+}
+
+/**
+ * @brief Function to verify presence and hence delete an entry from the trie
+ * @param entry the string entry to be deleted from the trie
+ */
+void Tnode ::Delete(std::string entry) {
+    Tnode *cur_pos = this,
+          *delete_from = this;  // Current pointer pointing to root
+    int letter_index = 0, delete_from_index = 0, i = 0, n = entry.size();
+
+    for (i = 0; i < n; i++) {
+        // To ignore case
+        letter_index = tolower(entry[i]) - 97;
+
+        // Display error message when given entry is not present in the tree
+        if (cur_pos->english[letter_index] == nullptr) {
+            std::cout << "Entry not Found" << std::endl;
+            return;
+        }
+        // If the current node is end of word for the current prefix or if it
+        // has 2 or more branches It cannot be deleted while deleting the
+        // required entry.
+        if (numberOfChildren(cur_pos) > 1 || cur_pos->endOfWord) {
+            delete_from = cur_pos;      // denotes the beginning of the shortest
+                                        // suffix that is allowed to be deleted
+            delete_from_index = i - 1;  // Beginning index of the suffix
+                                        // corresponding to the 'entry'
+        }
+
+        // Traversing through the entry
+        cur_pos = cur_pos->english[letter_index];
+    }
+
+    // cur_pos now points to the last char of entry. Display message if that
+    // entry does not exist
+    if (!cur_pos->endOfWord) {
+        std::cout << "Entry not Found" << std::endl;
+        return;
+    }
+
+    // If cur_pos is not a leaf node, unmark end of word and assign 0 to it's
+    // frequency for deletion
+    if (numberOfChildren(cur_pos)) {
+        cur_pos->endOfWord = false;
+        cur_pos->frequency = 0;
+        return;
+    }
+
+    // The first character of the suffix to be deleted
+    letter_index = tolower(entry[delete_from_index + 1]) - 97;
+    // Point cur_pos to the next node
+    cur_pos = delete_from->english[letter_index];
+    // Sever the connection from the main trie
+    delete_from->english[letter_index] = nullptr;
+
+    // If number of characters in the suffix are more than 1, recursively delete
+    // each character starting from cur_pos using the helper function
+    if (n > delete_from_index + 2) {
+        DeleteFrom(cur_pos, entry, delete_from_index + 2);
+    }
+    // If the suffix is only 1 char in length
+    else {
+        delete cur_pos;
+    }
+}
+
+/**
+ * @brief Function to check a word's presence in the trie (Basic)
+ * @param key the string key to be searched in the trie
+ * @return true if the key is found
+ * @return false if the key is not found
+ */
+bool Tnode ::SearchPresence(const std::string &key) {

```

Comment:
For better readability.
```suggestion
bool Tnode::SearchPresence(const std::string &key) {
```
Check out other parts of the code too. 🙂

Reply:
Corrected for all scope resolution operators.